src/compiler/checker.ts

```diff
@@ -29871,6 +29871,19 @@ namespace ts {
             }
         }
 
+        function isWellKnownSymbol(node: Node) {
+            if (!(
+                isPropertyAccessExpression(node) &&
+                isIdentifier(node.name) &&
+                isIdentifier(node.expression) &&
+                node.expression.escapedText === "Symbol"
+            )) {
+                return false;

```

Comment:
I don't like this. We're able to do way better than syntactic specialization. Instead, check if the type of the name expression is the same as one of the members of the global `SymbolConstructor` interface.